lxd/main_nsexec.go

```diff
@@ -681,6 +681,41 @@ void forkgetnet(char *buf, char *cur, ssize_t size) {
 	// The rest happens in Go
 }
 
+void proxydevstart(char *buf, char *cur, ssize_t size) {
+	int cmdline, listen_pid, connect_pid, fdnum;
+
+	// Get the arguments
+	ADVANCE_ARG_REQUIRED();
+	listen_pid = atoi(cur);
+	ADVANCE_ARG_REQUIRED();
+	ADVANCE_ARG_REQUIRED();
+	connect_pid = atoi(cur);
+	ADVANCE_ARG_REQUIRED();
+	ADVANCE_ARG_REQUIRED();
+	fdnum = atoi(cur);
+
+	// Cannot pass through -1 to runCommand since it is interpreted as a flag
+	fdnum = fdnum == 0 ? -1 : fdnum;
+	
+	char fdpath[80];
+	sprintf(fdpath, "/proc/self/fd/%d", fdnum);
```

Comment:
Some compilers don't like this and will spew a warning. I'd rather have this:

```
ret = snprintf(fdpath, sizeof(fdpath), "/proc/self/fd/%d", fdnum);
if (ret < 0 || (size_t)ret >= sizeof(fdpath))
        /* handle error */

The thing is that - although very rare - that `s{n}printf()` and friends can fail with `EIO`. The cast to `(size_t` is done to please compilers that would otherwise yell at us for comparing integers with different signs.